sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -0,0 +1,248 @@
+
+@Injectable()
+export class CASSupplierIntegrationService {
+  constructor(
+    private readonly casService: CASService,
+    private readonly systemUsersService: SystemUsersService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {}
+
+  /**
+   * CAS integration process.
+   * Logs on CAS supplier API and request the supplier information for the students with pending supplier information.
+   * @param parentProcessSummary parent process summary.
+   * @param casSuppliers pending CAS suppliers.
+   * @returns a number of update records.
+   */
+  async executeCASIntegrationProcess(
+    parentProcessSummary: ProcessSummary,
+    casSuppliers: CASSupplier[],
+  ): Promise<number> {
+    let suppliersUpdated = 0;
+    const summary = new ProcessSummary();
+    parentProcessSummary.children(summary);
+    try {
+      summary.info("Logging on CAS...");
+      const auth = await this.casService.logon();
+      if (auth.access_token) {
+        summary.info("Logon successful.");
+        suppliersUpdated = await this.requestCASAndUpdateSuppliers(
+          casSuppliers,
+          summary,
+          auth,
+        );
+      } else {
+        summary.info("Could not authenticate on CAS.");
+        throw new CustomNamedError(
+          "Could not authenticate on CAS.",
+          CAS_AUTH_ERROR,
+        );
+      }
+    } catch (error: unknown) {
+      summary.error("Unexpected error.", error);
+    }
+    return suppliersUpdated;
+  }
+
+  /**
+   * For each pending CAS supplier, request supplier information to CAS API and update local table.
+   * @param casSuppliers pending CAS suppliers.
+   * @param summary log summary.
+   * @param auth CAS auth details.
+   * @returns a number of update records.
+   */
+  private async requestCASAndUpdateSuppliers(
+    casSuppliers: CASSupplier[],
+    parentProcessSummary: ProcessSummary,
+    auth: CASAuthDetails,
+  ): Promise<number> {
+    let suppliersUpdated = 0;
+    for (const casSupplier of casSuppliers) {
+      const summary = new ProcessSummary();
+      parentProcessSummary.children(summary);
+      summary.info(`Requesting info for CAS supplier id ${casSupplier.id}.`);
+      let supplierResponse = null;
+      try {
+        try {
+          supplierResponse = await this.casService.getSupplierInfoFromCAS(
+            auth.access_token,
+            casSupplier.student.sinValidation.sin,
+            casSupplier.student.user.lastName.toUpperCase(),
+          );
+        } catch (error: unknown) {
+          throw new Error("Unexpected error while requesting supplier.", error);
+        }
+        suppliersUpdated = await this.updateActiveAddress(
+          supplierResponse,
+          summary,
+          casSupplier,
+          suppliersUpdated,
+        );
+      } catch (error: unknown) {
+        if (error instanceof Error) {
+          summary.error(error.message);
+        } else {
```

Comment:
Can you please clarify why we need to treat an `Error` differently?
Why the below would not be enough for both scenarios.
```ts
summary.error(
    "Error while requesting or updating CAS suppliers.",
    error,
);
```